lib/msal-node/src/cache/CacheManager.ts

```diff
@@ -0,0 +1,73 @@
+import { Storage } from './Storage';
+import { Serializer, Deserializer, InMemoryCache, JsonCache } from '@azure/msal-common';
+import { ICachePlugin } from './ICachePlugin';
+
+const defaultSerializedCache: JsonCache = {
+    Account: {},
+    IdToken: {},
+    AccessToken: {},
+    RefreshToken: {},
+    AppMetadata: {}
+};
+
+
+export class CacheManager {
+
+    hasChanged: boolean;
+    storage: Storage;
+    persistence: ICachePlugin;
+
+    constructor(storage: Storage) {
+        this.hasChanged = false;
+        this.storage = storage;
+        this.storage.registerChangeEmitter(this.handleChangeEvent.bind(this))
+
+    }
+
+    setPersistence(persistence: ICachePlugin): void {
+        this.persistence = persistence;
+    }
+
+    handleChangeEvent() {
+        this.hasChanged = true;
+    }
+
+    async serialize(): Promise<string> {
```

Comment:
I dont love serialize and deserialize, can have better namings open to suggestions

Reply:
leaving for now with the goal of possibly separating into two separate apis

Reply:
`serialize` and `deserialize` are used across MSAL libraries. The name is used mostly for users who are familiar with serialized persistence. Just FYI.